### PR TITLE
Initial version of the Quarkus stack

### PR DESCRIPTION
**Pull request by paulrobinson**
# Checklist:

- [x] Read the [Code of Conduct](https://github.com/appsody/website/blob/master/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md).

- [x] Followed the [commit message guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).

- [x] Stack adheres to [Appsody stack structure](https://github.com/appsody/website/blob/master/content/docs/stacks/stack-structure.md).

# Contributing a new stack:

## Describe how application dependencies are managed:
They are managed via the `properties` section of the `pom.xml` in the template code.

## Explain how Appsody file watcher is utilized:
I don't think it is used. I omitted the `APPSODY_WATCH_` variables as Quarkus will watch for change itself and perform live reload. Using the `APPSODY_MOUNTS` variable is enough to mount the developer's source code for Quarkus to watch.

## Describe other Appsody environment variables defined by the stack image:

`ENV APPSODY_RUN="./mvnw -Dmaven.repo.local=/mvn/repository compile quarkus:dev"`
This starts Quarkus in developer mode (live reload and remote debug enabled). It also configures Maven to use the `APPSODY_DEPS` location for the local repo cache. This allows maven to cache dependencies between container runs. 

`ENV APPSODY_DEBUG="./mvnw -Dmaven.repo.local=/mvn/repository -Ddebug compile quarkus:dev"`
This is the same as `APPSODY_RUN` but the Quarkus runtime pauses waiting for the debugger to attach.

`ENV APPSODY_TEST="./mvnw -Dmaven.repo.local=/mvn/repository test"`
This starts Quarkus and runs the tests. However, they currently fail. See 'known limitations' discussed below. 

## Describe any limitations and known issues:
I'm yet to test the `appsody deploy` command, as I don't yet have a Kubernetes cluster that's configured to deploy to.

There's also a problem with `appsody test`. The tests fail due to them not being able to communicate with the Quarkus server. It seems like the `8080` port is not being exposed.

I think both of these issues will need addressing before the PR is merged.

# Related Issues:
None.

### CODE REVIEW

**Comment by paulrobinson** [comment]
FYI: I will be on vacation, returning the 29th July. I can respond to any reviewing feedback when I return.

**Comment by neeraj-laad** [comment]
@paulrobinson Please can you review and sign the CLA. So we can progress this PR?

**Comment by paulrobinson** [comment]
I attempted to go through the process to sign the CLA. I thought it had
succeeded. I can try again when I return from holiday (29th July).

-- 
Get the Boomerang Email App <http://boomerangapp.com/mobile> on your phone



----------
Neeraj Laad <notifications@github.com> wrote:

@paulrobinson <https://github.com/paulrobinson> Please can you review and
sign the CLA. So we can progress this PR?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/appsody/stacks/pull/94?email_source=notifications&email_token=AAAMTYJOQTFLRA5KQJHBC5DQBGBNPA5CNFSM4IEPAZQ2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2ZEGWA#issuecomment-514999128>,
or mute the thread

File: incubator/quarkus/templates/default/.appsody-config.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# config file for application development
+stack: pfrobinson/quarkus-appsody-stack:latest
```
Comment (mhamwala) [inline incubator/quarkus/templates/default/.appsody-config.yaml]: can you change this is point to the appsody repo, example:
`appsody/java-quarkus:0.1`

**Comment by skoh7645** [comment]
Reviewed and tested:
 
- [x] appsody run
- [x] appsody test
- [x] appsody debug 
- [x] appsody build
- [x] appsody deploy

File: incubator/quarkus/templates/default/.appsody-config.yaml (hunk 1 above)
Comment (paulrobinson) [reply incubator/quarkus/templates/default/.appsody-config.yaml]: Is that better?

**Comment by paulrobinson** [comment]
I needed to amend one of the commits to ensure the right email address was being used, such that the CLA bot would validate my contribution. 

However, I'm still seeing an additional "Paul Robinson" in the CLAassistant list. I also seem to have messed up the merge and I'm seeing lots of extra commits that are not part of the PR.

I've created a tidied up PR with just my 2 commits in: https://github.com/appsody/stacks/pull/120

Closing this PR, in favour of using https://github.com/appsody/stacks/pull/120.

**Comment by paulrobinson** [comment]
I needed to amend one of the commits to ensure the right email address was being used, such that the CLA bot would validate my contribution. 

However, I'm still seeing an additional "Paul Robinson" in the CLAassistant list. I also seem to have messed up the merge and I'm seeing lots of extra commits that are not part of the PR.

I've created a tidied up PR with just my 2 commits in: https://github.com/appsody/stacks/pull/120

Closing this PR, in favour of using https://github.com/appsody/stacks/pull/120.

**Pull request closed without merging**